### PR TITLE
README update, clarification suggested by @mnieber

### PR DESCRIPTION
**Pull request by morisil**
@mnieber please take a look

### CODE REVIEW

**Comment by mnieber** [comment]
Yes, it's clear now! I read the rest of the code, and given your explanation of the event streams, I understand a bit better what this aims at. It still looks unfamiliar to me though, probably because I have never tried such an approach myself. But it's interesting!

File: README.md [hunk 1]
```diff
@@ -23,14 +24,26 @@ It's significant achievement when taking into account:
...
-Unlike in other cross-platform development methodologies, here UI developers are given full
-control over look and feel of their applications. I also believe that such an approach,
-abstracting application presentation logic from specific platform, brings much better overall
-architecture which pays off in the future.
+Just to name a few. Usually cross-platform development tools fall into these categories:
+
+* abstraction over native UI components and IO operations (specialized API
+accessible from any possible programming language which is later either interpreted
+or compiled for specific platform)
+* HTML+CSS forced as UI on mobile
```
Comment (mnieber) [inline README.md]: The use of HTML+CSS on all platforms, discarding the possibile advantages offered by native solutions on mobile devices.

File: README.md [hunk 2]
```diff
@@ -23,14 +24,26 @@ It's significant achievement when taking into account:
...
+accessible from any possible programming language which is later either interpreted
+or compiled for specific platform)
+* HTML+CSS forced as UI on mobile 
+
+But there is a third way, where only presentation logic code is shared and UI rendering
+and IO stays native. Where iOS, Web and Android developers can customize the app in every
+detail.
+
+This approach seems to be the most demanding one in terms of software architecture. Common business
+logic cannot operate on technical terms. Mouse click or touch event are becoming
```
Comment (mnieber) [inline README.md]: What does "operate on technical terms" mean? Do you mean that the business logic cannot be tightly coupled to UI events?

**Comment by morisil** [comment]
@mnieber regarding reactive programming and UI. I've already learned that RP is very useful for backend and completely changed the way I design software components now. I was also working with Android developers when they were introducing reactive paradigm for all the IO. But I had no idea that so called Functional Reactive Programming is reshaping frontend development maybe even more, until I found this article:

https://gist.github.com/staltz/868e7e9bc2a7b8c1f754

File: README.md (hunk 1 above)
Comment (morisil) [inline README.md]: rephrased

File: README.md (hunk 2 above)
Comment (morisil) [inline README.md]: I tried to explain it further

**Comment by mnieber** [comment]
This article is really interesting, I love it when people reinvent the way software is built. I just started reading, still have to read more about it :-)

**Comment by morisil** [comment]
@mnieber  thanks for your time. I hope to push changes to web demo soon. It already compiles, I just need to add material design. I will send you a link with the result. Then android version, I hope to make something there despite lack of hands on experience with the platform. And aferwards I need to buy a mac for iOS version. :) Actually I hope to find some iOS developers to help with this.

**Comment by mnieber** [comment]
Indeed I would like to take a look at the web client. The reactive approach is interesting, but my feeling so far is that I prefer to see more explicit code. In general, if prefer an approach like this one:

    class GithubUserListPanel:
        def __init__(self, loadGitHubUsers):
            self.loadGitHubUsers = loadGitHubUsers

        def onRefreshButtonClicked(self)
            self.loadGitHubUsers()

rather than have a refreshButtonClicked event that any client can subscribe to. Of course, this approach is much less flexible, but I find it easier to read. In an approach that connects several events and subscribers, I have to build a pipeline diagram in my head (or on paper) to see how things fit together. What I find attractive in the event-pipeline based approach is reduced boilerplate. It will be interesting to look at the web client, see how explicit it is, and think of ways to make it more explicit while still having the minimalistic features of the pipeline based approach.

**Comment by mnieber** [comment]
@morisil ^^^

**Comment by morisil** [comment]
@mnieber it will be something like this:

    public class DefaultUserView implements UserView {

      private final DivElement panel;
    
      private final SpanElement login;
    
      private final Observable<Trigger> selection$;
        
      public DefaultUserView() {
        Document doc = Browser.getDocument();
        panel = (DivElement) doc.querySelector("#userTemplate").cloneNode(true);
        login = (SpanElement) panel.querySelector(".login");
    
        selection$ = RxElemental.observe(panel, MOUSEDOWN)
            .mergeWith(RxElemental.observe(panel, TOUCHEND))
            .map(event -> Trigger.INSTANCE);
      }
    
      @Override
      public void displayLogin(String login) {
        this.login.setInnerText(login);
      }
    
      @Override
      public Observable<Trigger> observeSelection() {
        return selection$;
      }
    
      DivElement getPanel() {
        return panel;
      }
    
    }

The client code will be usually technical, as it is supposed to deal with technicalities and quirks of the platform, or even specific browser. That's the reason it is separated, to let people who are experts in fronted development deal with it, also in pure JavaScript.

BTW I checked the result of transpilation:

    function com_xemantic_githubusers_web_view_DefaultUserView_DefaultUserView__V(){
      var doc;
      doc = $doc;
      this.com_xemantic_githubusers_web_view_DefaultUserView_panel = doc.querySelector('#userTemplate').cloneNode(true);
      this.com_xemantic_githubusers_web_view_DefaultUserView_panel.querySelector('.login');
      new rx_Observable_Observable__Lrx_Observable$OnSubscribe_2V(
        rx_plugins_RxJavaHooks_onCreate__Lrx_Observable$OnSubscribe_2Lrx_Observable$OnSubscribe_2(
            new rx_internal_operators_OnSubscribeMap_OnSubscribeMap__Lrx_Observable_2Lrx_functions_Func1_2V(
                rx_Observable_merge__Lrx_Observable_2Lrx_Observable_2(
                    rx_Observable_from___3Ljava_lang_Object_2Lrx_Observable_2(
                        com_google_gwt_lang_Array_stampJavaTypeInfo__Ljava_lang_Class_2Lcom_google_gwt_core_client_JavaScriptObject_2Lcom_google_gwt_core_client_JavaScriptObject_2ILjava_lang_Object_2Ljava_lang_Object_2(
                            com_google_gwt_lang_Array_getClassLiteralForArray__Ljava_lang_Class_2ILjava_lang_Class_2(
                                com_google_gwt_lang_ClassLiteralHolder_Lrx_1Observable_12_1classLit, 1),
                                $intern_4, 13, 0,
                                [
                                    new rx_Observable_Observable__Lrx_Observable$OnSubscribe_2V(
                                        rx_plugins_RxJavaHooks_onCreate__Lrx_Observable$OnSubscribe_2Lrx_Observable$OnSubscribe_2(
                                            new rx_internal_operators_OnSubscribeCreate_OnSubscribeCreate__Lrx_functions_Action1_2Lrx_Emitter$BackpressureMode_2V(
                                                new com_intendia_rxgwt_elemental_RxElemental$lambda$1$Type_RxElemental$lambda$1$Type__Lelemental_events_EventTarget_2Ljava_lang_String_2ZV(
                                                    this.com_xemantic_githubusers_web_view_DefaultUserView_panel, 'mousedown'
                                                )
                                            )
                                        )),
                                    new rx_Observable_Observable__Lrx_Observable$OnSubscribe_2V(
                                        rx_plugins_RxJavaHooks_onCreate__Lrx_Observable$OnSubscribe_2Lrx_Observable$OnSubscribe_2(
                                            new rx_internal_operators_OnSubscribeCreate_OnSubscribeCreate__Lrx_functions_Action1_2Lrx_Emitter$BackpressureMode_2V(
                                                new com_intendia_rxgwt_elemental_RxElemental$lambda$1$Type_RxElemental$lambda$1$Type__Lelemental_events_EventTarget_2Ljava_lang_String_2ZV(
                                                    this.com_xemantic_githubusers_web_view_DefaultUserView_panel, 'touchend'
                                                )
                                            )
                                        )
                                    )
                                ]
                            )
                        )
                    )
                )
            )
      );
    }

The long method names are due to lack of namespaces in JavaScript, normally they are obfuscated to the shortest possible sequences of characters, except of those which are supposed to form API called from JavaScript. Still I am quite surprised how much inlining compiler is doing here. Maybe because Rx methods are usually single line of code.

**Pull request merged**